src/d3a/d3a_core/sim_results/file_export_endpoints.py

```diff
@@ -176,3 +146,74 @@ def _specific_row(self, slot, market):
             produced = self.area.strategy.state.get_energy_production_forecast_kWh(slot, 0.0)
             return [produced, not_sold]
         return []
+
+
+class FileExportEndpoints:
+    """Handle data preparation for csv-file and plot export."""
+    def __init__(self):
+        self.plot_stats = {}
+        self.plot_balancing_stats = {}
+        self.cumulative_offers = {}
+        self.cumulative_bids = {}
+        self.clearing = {}
+
+    def __call__(self, area):
+        self._populate_area_children_data(area)
+
+    def _populate_area_children_data(self, area: Area) -> None:
+        """Wrapper for calling self.update_plot_stats recursively"""
+        if area.children:
+            for child in area.children:
+                self._populate_area_children_data(child)
+        self._update_plot_stats(area)
+
+    @staticmethod
+    def export_data_factory(area: Area, past_market_type: PastDataEnum) -> ExportDataBase:
+        """Decide which data acquisition class to use."""
+        if past_market_type == PastDataEnum.SPOT_MARKET:
+            return ExportUpperLevelData(area.past_markets) \
+                if len(area.children) > 0 else ExportLeafData(area, area.past_markets)
+        if past_market_type == PastDataEnum.BALANCING_MARKET:
+            return ExportBalancingData(area.past_balancing_markets)
+        if past_market_type == PastDataEnum.SETTLEMENT_MARKET:
+            past_markets = list(area.past_settlement_markets.values())
+            return (ExportUpperLevelData(past_markets)
+                    if len(area.children) > 0 else ExportLeafData(area, past_markets))
+
+    def _get_stats_from_market_data(self, out_dict: Dict, area: Area,
+                                    past_market_type: PastDataEnum) -> None:
+        """Get statistics and write them to out_dict."""
+        data = self.export_data_factory(area, past_market_type)
+        if area.slug not in out_dict:
+            out_dict[area.slug] = dict((label, []) for label in data.labels)
+        for row in data.rows():
+            for ii, label in enumerate(data.labels):
+                out_dict[area.slug][label].append(row[ii])
+
+    def _populate_plots_stats_for_supply_demand_curve(self, area: Area) -> None:
+        """Populate the statistics for supply_demand_curve."""
+        if ConstSettings.IAASettings.MARKET_TYPE == 3:
```

Comment:
```suggestion
        if ConstSettings.IAASettings.MARKET_TYPE == 2 and ConstSettings.IAASettings.BID_OFFER_MATCH_TYPE == 2:
```

Reply:
good catch. This. was due to copy and paste.